src/common/http/platform/beast/http.cpp

```diff
@@ -255,6 +255,37 @@ size_t GetContentLength(const PARSER &parser) {
 	}
 }
 
+expected::ExpectedBool HasBody(
+	const expected::ExpectedString &content_length,
+	const expected::ExpectedString &transfer_encoding) {
+	bool chunked;
+	if (transfer_encoding) {
+		if (transfer_encoding.value() != "chunked") {
+			return expected::unexpected(error::Error(
+				make_error_condition(errc::not_supported),
+				"Unsupported Transfer-Encoding: " + transfer_encoding.value()));
+		}
+		chunked = true;
```

Comment:
Q: Why not simply return here?

Reply:
You're right, I guess my logic is a bit overkill. I'll do that.